thrift/thrift-gen/tchannel-template.go

```diff
@@ -118,14 +120,49 @@ func (s *{{ .ServerStruct }}) Service() string {
 func (s *{{ .ServerStruct }}) Methods() []string {
 	return []string{
 		{{ range .Methods }}
-			"{{ .ThriftName }}",
+			"{{ .ThriftName }}", 
 		{{ end }}
 		{{ range .InheritedMethods }}
 			"{{ . }}",
 		{{ end }}
 	}
 }
 
+// RegisterInterceptors registers the provided interceptors with the server.
+func (s *{{ .ServerStruct }}) RegisterInterceptors(interceptors ...thrift.Interceptor) {
+	if s.interceptors == nil {
+		interceptorsLength := len(interceptors)
+		s.interceptors = make([]thrift.Interceptor, 0, interceptorsLength)
+	}
+
+	s.interceptors = append(s.interceptors, interceptors...)
+}
+
+func (s *{{ .ServerStruct }}) callInterceptorsPre(ctx {{ contextType }}, method string, args athrift.TStruct) error {
```

Comment:
the templates should be as minimal as possible, all of the logic should live inside of the `thrift` package.